files/default/neutron-ha-tool.py

```diff
@@ -585,8 +589,58 @@ def migrate_router(qclient, router_id, agent, target):
     if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
                            (router_id, agent['id']))
+    if wait_for_ports:
+        wait_router_ports_active(qclient, router_id, target['host'])
 
 
+def wait_router_ports_active(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports of a router
+    for being in the ACTIVE status on a specific host
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    active_port_ids = set()
+    remaining = ['dummy']
+    LOG.info("Wait for the ports of router_id=%s to be ACTIVE" % router_id)
+    while (maxtries > 0 and len(remaining)):
+        # FIXME: Don't check the "network:router_interface_distributed" ports
+        # as those are not hosted on the network node
+        router_port_list = qclient.list_ports(device_id=router_id,
+                                              fields=['id',
+                                                      'status',
+                                                      'binding:host_id'])
+        ports_to_check = [port for port in router_port_list['ports']
+                          if port['id'] not in active_port_ids]
+        remaining = []
+        for port in ports_to_check:
+            LOG.debug("Router port=%s on hostid=%s in status=%s" %
+                      (port['id'], port['binding:host_id'],
+                       port['status']))
+            if (port['binding:host_id'] == target_host and
+                port['status'] == 'ACTIVE'):
+                LOG.info("port_id=%s is ACTIVE on host_id=%s" %
+                         (port['id'], target_host))
+                active_port_ids.add(port['id'])
+            else:
+                remaining.append(port['id'])
```

Comment:
It's not that easy, actually. The list of ports needs to be refetched with every iteration, as the port state will change. And they might even change in number (e.g. users might have added/delete ports meanwhile).
Additionally (e.g. for logging/debugging purposes) I'd like to keep the state for which ports are already completed. While avoiding to log the over and over again. If I adjust your above suggestion to include those two things I think the resulting code has much of an advantage over  the current code.